spec/views/collections/_show_document_list_menu.html.erb_spec.rb

```diff
@@ -1,18 +1,35 @@
 
 describe 'collections/_show_document_list_menu.html.erb', type: :view do
-  context 'when user logged in displaying the collections of current user' do
+  context 'when user is viewing a collection' do
     let(:user) { create :user }
-    let(:document) { SolrDocument.new(id: '1234') }
+    let(:another_user) { create :user }
+    let(:document) { SolrDocument.new(id: '1234', edit_access_person_ssim: [user.email]) }
```

Comment:
Instead of stubbing access on the document, try this: `allow(view).to receive(:can?).with(:edit, document).and_return(true/false)`